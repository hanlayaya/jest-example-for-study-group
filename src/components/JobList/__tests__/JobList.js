import React from 'react';
import { mount } from 'enzyme';

describe('<JobList />', function () {
  let JobList;

  beforeEach(() => {
    jest.resetModules();
    JobList = require('../JobList').default;

    this.params = {};

    this.makeSubject = () => {
      const { jobs } = this.params;

      return mount(<JobList jobs={jobs} />);
    };
  });

  describe('when it has no any data', () => {
    beforeEach(() => {
      this.params.jobs = [];

      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(this.subject.html()).toMatchSnapshot();
    });
  });

  describe('when it has some jobs data', () => {
    beforeEach(() => {
      this.params.jobs = [{
        id: 3,
        title: 'Frontend Developer',
        company: 'Faceb00k',
      }, {
        id: 2,
        title: 'Backend Developer',
        company: 'Goog1e',
      }, {
        id: 1,
        title: 'UI Designer',
        company: '4irBnB',
      }];

      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(this.subject.html()).toMatchSnapshot();
    });

    it('should display `(unknown)`', () => {
      expect(this.subject.find('h1').length).toEqual(3);
    });
  });
});
