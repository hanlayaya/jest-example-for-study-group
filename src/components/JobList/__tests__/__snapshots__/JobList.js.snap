// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobList /> when it has no any data should be same as snapshot 1`] = `"<div>No data</div>"`;

exports[`<JobList /> when it has some jobs data should be same as snapshot 1`] = `"<div><div><h1>Frontend Developer</h1><h2>Faceb00k</h2></div><div><h1>Backend Developer</h1><h2>Goog1e</h2></div><div><h1>UI Designer</h1><h2>4irBnB</h2></div></div>"`;
